tests/integration/data/test-bundle-ceph.yaml

```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: integration-test-ceph
+description: |-
+  Used to deploy or refresh within an integration test model
+series: jammy
+applications:
+  k8s:
+    charm: k8s
+    channel: latest/edge
+    constraints: cores=2 mem=8G root-disk=16G
+    num_units: 1
+  k8s-worker:
+    charm: k8s-worker
+    channel: latest/edge
+    constraints: cores=2 mem=8G root-disk=16G
+    num_units: 1
+  ceph-csi:
+    charm: ceph-csi
+    channel: latest/stable
+    options:
+      provisioner-replicas: 1
```

Comment:
this is a subordinate charm and it cannot ever have constraints attached.  Notice that `num_units` isn't set.